python/singa/autograd.py

```diff
@@ -1828,3 +1828,27 @@ def backward(self, dy):
 
 def leakyrelu(x, a=0.01):
     return LeakyRelu(a)(x)[0]
+
+
+class SoftPlus(Operation):
+    def __init__(self):
+        super(SoftPlus, self).__init__()  
+    
+    def forward(self, x):
+    #f(x) = ln(exp(x) + 1)
+        if training:
+            self.input = x
+        x1 = singa.AddFloat(singa.Exp(x),1.0)
+        y = singa.Log(x1)
+        
+        return y
+​
+    def backward(self, dy):
+        dx = singa.AddFloat(singa.Exp(self.input),1.0)
```

Comment:
please use the formula: dx = 1 / (1 + exp(-x)). Your implement here has to compute the exp(x) twice.

Reply:
yes. backward function changed.